ts/src/base/Exchange.ts

```diff
@@ -2837,6 +2837,10 @@ export default class Exchange {
         return cleanStructure;
     }
 
+    isSpotMargin (market: MarketInterface, marketType: Str = undefined, marginMode: Str = undefined): boolean {
+        return market['margin'] || (marketType === 'margin') || ((marketType === 'spot' && marginMode !== undefined))
```

Comment:
@ttodua we can't do this, `market['margin']` is a property that indicates if the market supports margin or not, it does not mean th euser wants to use it,

Reply:
let me recheck

Reply:
uh oh, yes i see

Reply:
pushed

Reply:
so, what i see in binance's createOrder, we do support `type` parameter, which seems redundant, but per my guess it's because we support conflicting market-ids (as symbols), so we support backward users like: 
```
symbol = 'BTCUSDT`, type = `margin`
```
if that's case, then i've also pushed some edits that makes market-variables more specific to be used `if` clauses